docs/protocol/specifications/blockchain.md

```diff
@@ -0,0 +1,613 @@
+
+Witness field       | Type                 | Description
+--------------------|----------------------|----------------
+Destination         | ValueDestination1    | The Destination ("forward pointer") for the value contained in this spend. This can point directly to an Output entry, or to a Mux, which points to Output entries via its own Destinations.
+Arguments           | List<String>         | Arguments for the control program contained in the SpentOutput.
+AnchoredEntry       | Pointer                      | Optional pointer to a single entry of any type, which uniquely identifies that entry as one that can use this one as an `Anchor`.
+
+#### Spend Validation
+
+1. Verify that `SpentOutput` is present in the transaction, but do not validate it.
+2. [Validate program](#program-validation) `SpentOutput.ControlProgram` with the given `Arguments` and the transaction version.
+3. Verify that `SpentOutput.Value` is equal to `Destination.Value`.
+4. [Validate](#value-destination-1-validation) `Destination`.
+5. If the transaction version is 1: verify that the `ExtHash` is the all-zero hash.
+
+### Issuance 1
+
+Field               | Type                 | Description
+--------------------|----------------------|----------------
+Type                | String               | "issuance1"
+Body                | Struct               | See below.
+Witness             | Struct               | See below.
+
+Body field          | Type                            | Description
+--------------------|---------------------------------|----------------
+Anchor              | Pointer<Nonce|Spend1|Issuance1> | Entry that this issuance is anchored to.
+Value               | AssetAmount1           | Asset ID and amount being issued.
+Data                | Hash                   | Hash of the reference data for this entry, or a string of 32 zero-bytes (representing no reference data).
+ExtHash             | [ExtStruct](#extension-struct)| If the transaction version is known, this must be 32 zero-bytes.
+
+Witness field       | Type                                      | Description
+--------------------|-------------------------------------------|----------------
+Destination         | ValueDestination1                         | The Destination ("forward pointer") for the value contained in this spend. This can point directly to an `Output`, or to a `Mux`, which points to `Output` entries via its own `Destinations`.
+AssetDefinition     | [Asset Definition](#asset-definition)     | Asset definition for the asset being issued.
+Arguments           | List<String>                              | Arguments for the control program contained in the SpentOutput.
+AnchoredEntry       | Pointer                                   | Optional pointer to a single entry of any type, which uniquely identifies that entry as one that can use this one as an `Anchor`.
+
+#### Issuance Validation
+
+**Inputs:**
+
+1. Issuance entry,
+2. initial block ID.
```

Comment:
I agree with the principle, but not sure if that would lead to an overkill. Could we see organizing that data as an implementation detail, irrelevant to the logic of the spec?